Src/Sarif.PatternMatcher/SearchSkimmer.cs

```diff
@@ -155,6 +155,12 @@ public override void Analyze(AnalyzeContext context)
             {
                 try
                 {
+                    var fileInfo = new FileInfo(filePath);
+                    if (context.FileSizeInKilobytes != -1 && fileInfo.Length / 1024 > context.FileSizeInKilobytes)
```

Comment:
>if (context.FileSizeInKilobytes != -1 && fileInfo.Length / 1024 > context.FileSizeInKilobytes) [](http://example.com/codeflow?start=20&length=94)

you would need a similar condition to the else clause.

The file is already in memory and you should return if the size of it is greater than the max allowed.